crates/proto/src/tests/tcp.rs

```diff
@@ -85,7 +85,11 @@ fn tcp_server_setup(
 }
 
 /// Test tcp_stream.
-pub fn tcp_stream_test<S: Connect, E: Executor, TE: Time>(server_addr: IpAddr, mut exec: E) {
+pub fn tcp_stream_test<R: RuntimeProvider, E: Executor, TE: Time>(
+    server_addr: IpAddr,
+    mut exec: E,
+    runtime: R,
```

Comment:
For tests like this, which take a `runtime: RuntimeProvider` arg and then just pass it to e.g. `TcpStream::new`, could we instead either require that the `RuntimeProvider` type is `Default`, like you do here: https://github.com/bluejekyll/trust-dns/blob/9a03a7592e7e011ad544188540c6718ab07b38d9/crates/proto/src/tests/udp.rs#L12

So that the test can take care of creating `runtime` itself, or even require that all implementations of `RuntimeProvider` themselves implement `Default`?

Reply:
I'm generally against requiring `Default` for the `RuntimeProvider` trait, since some `RuntimeProvider` implementations may not have a reasonable `Default` implementation, Additionally,  if the `RuntimeProvider` requries a `Default` implementation, it'd be easy to implement a transport that should really be receiving the user supplied `RuntimeProvider` instance but is instead constructing it's own instance.

However, I'm OK with these tests requiring `Default`, but it would just force the caller to use the turbofish operator to specify the runtime provider instead of passing it through as a function parameter.

Reply:
That's fair, I don't think `RuntimeProvider` should necessarily require `Default`.

> it would just force the caller to use the turbofish operator to specify the runtime provider instead of passing it through as a function parameter.

IIUC most (all?) instances of these tests already have to use the turbofish to specify the `Executor` and `Time` type parameters; since most implementations of `RuntimeProvider` are stateless, I think it's more symmetric for a test to look like this:

```
tcp_stream_test::<MyRuntime, MyExecutor, MyTime>(...)
```

than this: 
```
tcp_stream_test::<_, MyExecutor, MyTime>(..., MyRuntime)
```

Since, after all, the type parameter still has to be there even if it is elided. Anyway, I don't feel super strongly about this, just a suggestion.